app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt

```diff
@@ -0,0 +1,19 @@
+package au.com.gridstone.trainingkotlin.data
+
+import com.squareup.moshi.JsonClass
+
+// mapping jason top object
+@JsonClass(generateAdapter = true)
```

Comment:
you can avoid all the JsonAdapter generation, if we use the `KotlinJsonAdapter()`
```kotlin
Moshi.Builder()
  .add(KotlinJsonAdapterFactory())
  .build() 
```

Reply:
Sure